File: python/examples/smoke.py
#!/usr/bin/env python3
"""Live smoke test for the ``stockseyes`` Python SDK.

Build/install first, then run with your key::

    pip install -e ".[dev]"
    STOCKSEYES_RAPIDAPI_KEY=your_key python examples/smoke.py
"""

from __future__ import annotations

import os
import sys


def main() -> None:
    from stockseyes import (
        StockEyesConfig,
        is_stockeyes_error,
        use_stockeyes,
    )

    api_key = os.environ.get("STOCKSEYES_RAPIDAPI_KEY", "")
    if not api_key:
        print("Set STOCKSEYES_RAPIDAPI_KEY to your RapidAPI key first.", file=sys.stderr)
        sys.exit(1)

    client = use_stockeyes(StockEyesConfig(api_key=api_key))

    try:
        print('quote("RELIANCE")…')
        q = client.quote("RELIANCE")
        print(f"  {q.symbol}  ₹{q.price}  ({q.change_percent:.2f}%)  @ {q.exchange}")

        print('search("REL")…')
        s = client.search("REL")
        for i in s.results:
            print(f"  {i.symbol} — {i.name}")

        print("\n✅ Live smoke test passed.")

    except Exception as err:
        if is_stockeyes_error(err):
            print(
                f"\n❌ StockEyesError [{err.code}] status={err.status}: {err}",  # type: ignore[attr-defined]
                file=sys.stderr,
            )
        else:
            print(f"\n❌ Unexpected error: {err}", file=sys.stderr)
        sys.exit(1)


if __name__ == "__main__":
    main()

File: kotlin/examples/smoke.kt
package com.stockseyes.examples

import com.stockseyes.StockEyesConfig
import com.stockseyes.StockEyesError
import com.stockseyes.isStockEyesError
import com.stockseyes.useStockEyes
import kotlin.system.exitProcess

fun main() {
    val apiKey = System.getenv("STOCKSEYES_RAPIDAPI_KEY") ?: ""
    if (apiKey.isEmpty()) {
        System.err.println("Set STOCKSEYES_RAPIDAPI_KEY to your RapidAPI key first.")
        exitProcess(1)
    }

    val client = useStockEyes(StockEyesConfig(apiKey = apiKey))

    try {
        println("quote(\"RELIANCE\")…")
        val q = client.quote("RELIANCE")
        println("  ${q.symbol}  ₹${q.price}  (${String.format("%.2f", q.changePercent)}%)  @ ${q.exchange}")

        println("search(\"REL\")…")
        val s = client.search("REL")
        for (i in s.results) {
            println("  ${i.symbol} — ${i.name}")
        }

        println("\n✅ Live smoke test passed.")
    } catch (err: Throwable) {
        if (err is StockEyesError) {
            System.err.println("\n❌ StockEyesError [${err.code}] status=${err.status}: ${err.message}")
        } else if (isStockEyesError(err)) {
            System.err.println("\n❌ StockEyesError (via type guard): ${err.message}")
        } else {
            System.err.println("\n❌ Unexpected error: ${err.message}")
            err.printStackTrace()
        }
        exitProcess(1)
    }
}
